semantic_router/layer.py

```diff
@@ -216,15 +216,13 @@ def __init__(
         for route in self.routes:
             if route.score_threshold is None:
                 route.score_threshold = self.score_threshold
+
+        if self.routes:
+            self._add_routes(routes=self.routes)
```

Comment:
this logic seems incorrect, it looks like we try adding routes via `self._add_routes` then a few lines down we may be adding them again with `self._add_and_sync_routes`

@Vits-99 could you also review this section to confirm the functionality?

Reply:
Why just not modifying the sync method to accept also function schemas? Or you could associate those function_schemas to route_names and after having synced add that information to the pinecone upsert as metadata associating those function schemas back to the routes to add output from sync method